File: ICA0011/BeautifulSoap/soap_client.php
<?php

error_reporting(E_ALL);
ini_set('display_errors', 'On');
header('Content-Type: application/json');

$client = new SoapClient('http://localhost:8000/?wsdl');

$disk = $client->disk_usage();
$dns = $client->dns_records(array('name' => 'google.com'));
$user_info = $client->get_information(array('reachable_host' => 'google.com'));

echo json_encode([$user_info, $dns, $disk], JSON_PRETTY_PRINT);

?>

File: ICA0011/BeautifulSoap/soap_client.py
from suds.client import Client

client = Client('http://localhost:8000/?wsdl', cache=None)

print("Service #1")
user_info = client.service.get_information("www.google.com")
# print(user_info)
print('\n'.join(user_info[0]))
print()

print("Service #2")
dns = client.service.dns_records("twitter.com")
# print(dns)
print('\n'.join(dns[0]))
print()

print("Service #3")
disk = client.service.disk_usage()
# print(disk)
print('\n'.join(disk[0]))
print()

print("Service #4")
block = client.service.block_domain("facebook.com", "unblock")
# print(block)
print('\n'.join(block[0]))

